korangar/src/loaders/animation/mod.rs

```diff
@@ -0,0 +1,474 @@
+                            Some(value) => value == 1,
+                            None => false,
+                        };
+                        if entity_type == EntityType::Player && has_attach_point && animation_index == 1 {
+                            let parent_animation_pair = &vec[0];
+                            let parent_action = &parent_animation_pair.actions.actions[action_index];
+                            let parent_motion = &parent_action.motions[motion_index];
+                            let parent_attach_point = parent_motion.attach_points[0].position;
+                            let attach_point = motion.attach_points[0].position;
+                            let new_offset = -attach_point + parent_attach_point;
+                            offset += new_offset;
+                        }
+                        let size = Vector2::new(width as i32, height as i32);
+
+                        frame_part = FramePart {
+                            animation_index: animation_index as i32,
+                            sprite_number,
+                            size,
+                            upleft: Vector2::zero(),
+                            offset,
+                            mirror,
+                            angle,
+                            color,
+                        };
+                        let frame = Frame {
+                            size,
+                            upleft: Vector2::zero(),
+                            offset,
+                            remove_offset: Vector2::zero(),
+                            frameparts: vec![frame_part],
+                        };
+                        motion_frames.push(frame);
+                    }
+                    if motion_frames.len() == 1 {
+                        action_frames.push(motion_frames[0].clone());
+                    } else {
+                        let frame = Frame::merge_frame_part(&mut motion_frames);
+                        action_frames.push(frame);
+                    }
+                    motion_index += 1;
+                }
+                animation_frames.push(action_frames);
+                action_index += 1;
+            }
+            animations_list.push(animation_frames);
+            animation_index += 1;
+        }
+        let action_size = vec[0].actions.actions.len();
+        let animation_pair_size = vec.len();
+
+        let mut animations: Vec<Animation> = Vec::new();
+
+        // Generate for each action, the several motions that it have and create a
+        // offset of retangles.
+        for action_index in 0..action_size {
+            let motion_size = vec[0].actions.actions[action_index].motions.len();
+            let mut frames: Vec<Frame> = Vec::new();
+            for motion_index in 0..motion_size {
+                let mut generate: Vec<Frame> = Vec::new();
+
+                for animation_pair_index in 0..animation_pair_size {

```

Comment:
No need to iter using the index, you can iter over animations_list directly.

Reply:
I am trying to find ways to directly iterate, hasn't found any ways yet.

Reply:
Example:

```
                for pair in &animations_list[0..animation_pair_size] {
                    if pair.len() <= action_index {
                        continue;
                    }
                    if pair[action_index].len() <= motion_index {
                        continue;
                    }
                    generate.push(pair[action_index][motion_index].clone());
                }
```

But it seems you can iterate over the full animations_list too.

Reply:
Nice catch